### PR TITLE
[SW-2146] Cleanup in tests -> move to right ai.h2o.sparkling packages

### PR DESCRIPTION
**Pull request by jakubhava**
Last big shuffle with classes, this time with tests classes only:

-> Moving all remaining classes to ai.h2o.sparkling package
-> Putting TestUtils to a single class test utils
-> Extract ml integ tests from core and put them into ml integ tests
-> Few other extractions -> just moving tests to packages where they belong

Wanted to do this change now as we are getting closer to major release as we will switch rel branch to current master soon and avoid possible cherry-pick issues later.

**No tests were removed** ( edit: except 2 in SupportedRDDTestSuite -> those tests are useless now)

### CODE REVIEW

File: core/src/test/scala/ai/h2o/sparkling/backend/converters/SW516StandaloneApp.scala [hunk 1]
```diff
@@ -1,73 +0,0 @@
-/*
```
Comment (jakubhava) [inline core/src/test/scala/ai/h2o/sparkling/backend/converters/SW516StandaloneApp.scala:1]: Unused class

File: core/src/test/scala/org/apache/spark/h2o/utils/H2OAsserts.scala [hunk 2]
```diff
@@ -1,69 +0,0 @@
-/*
```
Comment (jakubhava) [inline core/src/test/scala/org/apache/spark/h2o/utils/H2OAsserts.scala:1]: Moved to ai.h2o.sparkling.TestUtils

File: core/src/test/scala/org/apache/spark/h2o/testdata/TestData.scala [hunk 3]
```diff
@@ -1,114 +0,0 @@
-/*
```
Comment (jakubhava) [inline core/src/test/scala/org/apache/spark/h2o/testdata/TestData.scala:1]: Moved to ai.h2o.sparkling.TestUtils

File: ml/src/test/scala/ai/h2o/sparkling/ml/PipelinePredictionTest.scala [hunk 4]
```diff
@@ -31,7 +31,6 @@ import org.apache.spark.{SparkContext, SparkFiles}
 import org.junit.runner.RunWith
 import org.scalatest.FunSuite
 import org.scalatest.junit.JUnitRunner
-import water.api.TestUtils
 
 abstract class PipelinePredictionTestBase extends FunSuite with SharedH2OTestContext {
```
Comment (mn-mikke) [inline ml/src/test/scala/ai/h2o/sparkling/ml/PipelinePredictionTest.scala:35]: nit: Could you rename the file to match the conventions: `PipelinePredictionTest` -> `PipelinePredictionTestSuite`?

File: ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OKMeansTest.scala [hunk 5]
```diff
@@ -17,14 +17,13 @@
...
-import org.apache.spark.h2o.utils.SharedH2OTestContext
 import org.apache.spark.ml.{Pipeline, PipelineModel}
 import org.apache.spark.sql.Row
 import org.junit.runner.RunWith
 import org.scalatest.junit.JUnitRunner
 import org.scalatest.{FunSuite, Matchers}
-import water.api.TestUtils
 
 @RunWith(classOf[JUnitRunner])
 class H2OKMeansTest extends FunSuite with Matchers with SharedH2OTestContext {
```
Comment (mn-mikke) [inline ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OKMeansTest.scala:29]: nit: `H2OKMeansTest` -> `H2OKMeansTestSuite`

File: ml/src/test/scala/ai/h2o/sparkling/ml/PipelinePredictionTest.scala (hunk 4 above)
Comment (jakubhava) [reply ml/src/test/scala/ai/h2o/sparkling/ml/PipelinePredictionTest.scala:35]: Will do

File: ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OKMeansTest.scala (hunk 5 above)
Comment (jakubhava) [reply ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OKMeansTest.scala:29]: Will do

**Pull request merged**